app/services/document_converter_service.rb

```diff
@@ -0,0 +1,43 @@
+class DocumentConverterService
+  def initialize(original, converted)
+    @original = original
+    @converted = converted
+  end
+
+  def call
+    return if @converted.attached?
+
+    if @original.content_type == 'application/pdf'
+      @converted.attach(@original.blob)
+    else
+      convert_to_pdf from: @original, to: @converted
+    end
+  end
+
+  private
+
+  def convert_to_pdf(from:, to:)
+    with_attached_file(from) do |original|
+      pdf_tmpfile = Tempfile.new
+      Libreconv.convert(original, pdf_tmpfile)
+      to.attach(io: pdf_tmpfile, filename: "#{from.filename}.pdf", content_type: 'application/pdf')
+    end
+  rescue IOError
+    nil # raised if Libreoffice exe is not in PATH
+  end
+
+  def with_attached_file(document)
+    # Currently, evidence documents are converted to PDF when the document is
+    # uploaded and before the instance of Document is saved. This is how it
+    # was done with Paperclip but with Active the file is not accessible at
+    # until it is saved. This is why the path is taken as the temporary file
+    # of the attachment.
```

Comment:
Seems a bit of an excessive comment that will go out of date quite quickly. better that this stuff is in commit message history but I think the meat of it is "AS files are not readable/accessible until saved"